department-of-veterans-affairs/caseflow
# MattT/APPEALS-8403: Remove :vha_predocket_appeals FeatureToggle

Resolves [APPEALS-8403](https://vajira.max.gov/browse/APPEALS-8403)

## Description
This PR removes use of the `vha_predocket_appeals` FeatureToggle as it no longer serves a purpose within Caseflow.

**Note:** There is an accompanying PR in the appeals-deployment repository that can be merged after this PR is merged: https://github.com/department-of-veterans-affairs/appeals-deployment/pull/3726

## Acceptance Criteria
- [ ] All references to `vha_predocket_appeals` are removed from the Caseflow codebase
- [ ] All references to `vha_predocket_appeals` are removed from [appeals-deployment/ansible/vars/functions-config.yml](https://github.com/department-of-veterans-affairs/appeals-deployment/blob/master/ansible/vars/functions-config.yml)

## Testing Plan
1. Run `make enable-feature-flags > feature-flags-output`. This will clear all enabled FeatureToggles and enable the ones the script finds references to in the code. It will save the output of this command in a file named `feature-flags-output`.

2. Make sure there is no mention of `vha_predocket_appeals` in `feature-flags-output`:

(Don't include the $ signs. They just denote a line in a non-root shell.)
```bash
$ cat feature-flags-output | grep "vha_predocket_appeals"
$
```

3. Delete `feature-flags-output`:

```bash
$ rm feature-flags-output
```

4. Log into Caseflow as an Intake user (ex: `BVAISHAW`).

5. Proceed through the intake process for a 10182 appeal.

6. On the Add/Remove Issues page, add an issue with a benefit type of "Veterans Health Administration". The issue category that you select doesn't matter. Make sure this banner pops up on the modal:
<img width="450" alt="vha-issue-banner" src="https://user-images.githubusercontent.com/99351305/185478464-3a9aaa4c-ce42-4d34-98ed-e819ff39c84f.PNG">

7. Click "Add this Issue". You should now see this banner on the Add/Remove Issues page:
<img width="751" alt="add-remove-issues-banner" src="https://user-images.githubusercontent.com/99351305/185478605-e07a7ee8-f63e-445e-8697-7edabef9509c.PNG">

8. Click "Submit appeal".

9. Ensure that this text appears on the completion page:
<img width="740" alt="vha-completion-message" src="https://user-images.githubusercontent.com/99351305/185478720-34afdd7d-6cf7-4b0f-8a20-5d4d38ba6285.PNG">

10. Go to the Case Details page for the appeal (`/queue/appeals/<appeal-UUID>`). Make sure a PreDocketTask has been created for the appeal:
<img width="508" alt="vha-predocket-task" src="https://user-images.githubusercontent.com/99351305/185478888-ab2e7a62-961f-4958-ae13-ba11193313d7.PNG">

11. If you'd like further confirmation, you can run `Appeal.find_by_uuid("<appeal-UUID").treee` in the console, and see that the task tree appears as it does below:

```ruby
                                      ┌──────────────────────────────────────────────────────────────────┐
Appeal 1637 (D 220731-1637 Original)  │ ID   │ STATUS   │ ASGN_BY  │ ASGN_TO   │ UPDATED_AT              │
└── RootTask                          │ 7111 │ on_hold  │          │ Bva       │ 2022-08-18 19:15:07 UTC │
    └── PreDocketTask                 │ 7112 │ on_hold  │          │ BvaIntake │ 2022-08-18 19:15:07 UTC │
        └── VhaDocumentSearchTask     │ 7113 │ assigned │ BVAISHAW │ VhaCamo   │ 2022-08-18 19:15:07 UTC │
                                      └──────────────────────────────────────────────────────────────────┘
```
